src/NUnitEngine/nunit.engine/Runners/ProcessRunner.cs

```diff
@@ -249,7 +250,15 @@ protected override void Dispose(bool disposing)
                     }
                     catch (SocketException se)
                     {
-                        var exitCode = _agency.GetAgentExitCode(_agent.Id);
+                        int? exitCode;
+                        try
+                        {
+                            exitCode = process.ExitCode;
+                        }
+                        catch (NotSupportedException)
```

Comment:
Nice future-proofing 👍 (I presume!)

Reply:
The try...catch was present before I moved this code. 

https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.exitcode#exceptions:

> NotSupportedException
> You are trying to access the ExitCode property for a process that is running on a remote computer. This property is available only for processes that are running on the local computer.

I've also run into an exception when accessing the ExitCode property for an exited process that I didn't start, but I don't remember which exception it was. NotSupportedException sounds familiar though.